webapp/channels/src/components/post_view/message_attachments/action_button/action_button.tsx

```diff
@@ -21,8 +21,15 @@ type Props = {
     actionExecutingMessage?: string;
 }
 
-export default class ActionButton extends React.PureComponent<Props> {
-    getStatusColors(theme: Theme) {
+const ActionButton = ({
+    action,
+    handleAction,
+    disabled,
+    theme,
+    actionExecuting,
+    actionExecutingMessage,
+}: Props) => {
+    const getStatusColors = (theme: Theme) => {
```

Comment:
This doesn't depend on anything inside the component, so let's move it outside of the component.